apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/DslJsonSerializer.java

```diff
@@ -965,7 +965,7 @@ private void serializeHttpContext(final Http http) {
             if (statusCode > 0) {
                 writeField("status_code", http.getStatusCode());
             }
-            writeLastField("url", http.getUrl());
+            writeLastField("url", http.getUrl().getFull());
```

Comment:
I wonder if we need to call `fillFullUrl` or `updateFull` to avoid the `url` to be serialized to an empty string. Do we have tests for that case?

Reply:
We don't have a test for that, and I wonder if we could simply throw an `IllegalStateException` when calling `getFull` when the `full` attribute is empty to indicate those errors:
- individual properties have been properly set, but `updateFull()` has not
- there was no call to `withFull` with a raw string value